File: 0057-Insert-Interval/lc0057.py
# Python solution.

class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        s, e = newInterval
        lstl, lstr = [], []

        for l, r in intervals:
            if s > r:
                lstl += [[l, r]]
            elif e < l:
                lstr += [[l, r]]
            else:
                s, e = min(s, l), max(e, r)

        return lstl + [[s, e]] + lstr

File: 0057-Insert-Interval/lc0057.rs
// Rust solution.

impl Solution {
    pub fn insert(intervals: Vec<Vec<i32>>, new_interval: Vec<i32>) -> Vec<Vec<i32>> {
        let (mut start, mut end) = (new_interval[0], new_interval[1]);
        let mut lstl: Vec<Vec<i32>> = Vec::new();
        let mut lstr: Vec<Vec<i32>> = Vec::new();

        for interval in intervals.into_iter() {
            let (left, right) = (interval[0], interval[1]);
            if start > right {
                lstl.push(interval);
            } else if end < left {
                lstr.push(interval);
            } else {
                start = start.min(left);
                end = end.max(right);
            }
        }

        lstl.push(vec![start, end]);
        lstl.extend(lstr);
        lstl
    }
}
